definitions/features/katello.rb

```diff
@@ -14,4 +14,29 @@ def data_dirs
   def current_version
     @current_version ||= rpm_version('katello')
   end
+
+  def services
+    #TODO split into katello/foreman services
+    # services have a number for priority in order to ensure
+    # they are started and stopped in the correct order.
+    {
+      'mongod'                   => 5,
+      'postgresql'               => 5,
+      'qpidd'                    => 10,
+      'qdrouterd'                => 10,
+      'squid'                    => 10,
+      #'tomcat'                   => 20,
+      'tomcat6'                  => 20,
+      'pulp_workers'             => 20,
+      'pulp_celerybeat'          => 20,
+      'pulp_resource_manager'    => 20,
+      'pulp_streamer'            => 20,
+      'foreman-proxy'            => 20,
+      'smart_proxy_dynflow_core' => 20,
+      'httpd'                    => 30,
+      #'puppetserver'             => 30,
+      'foreman-tasks'            => 30,
+      'goferd'                   => 30
+    }
+  end
```

Comment:
@iNecas @mbacovsky Should I split up these services into different features? They are being selected based if they [exist on the sytem](https://git.io/vADZI) We could keep one master list of services and it shouldn't make a difference if its standalone foreman/foreman-proxy/etc. Curious to hear your thoughts.

Reply:
If we do go with one list, I also am wondering where the best home for it would be :smile:

Reply:
What about every relevant feature (foreman, katello, pulp, tasks…) including some module (something like FeatureWithServices) and providing a method (e.g. `services_info`) that would provide the partial hash for it's part.  Alternatively to including some module, we could mark those features with some metadata tag.

We could then get the list of relevant services with something like `features.find_all { |f| f.is_a? FeatureWithServices }.map(&:service_info).reduce(&:merge)`.

WIth this approach, I believe we can achieve better modularity of the scripts

Reply:
I like the idea of moving the services into the features. I used the same approach with config files in backup (each feature with configs has method config_files returning the list). 
I also like the the idea of tagging the features and putting the master list together dynamically. For the config files the `HasConfigFiles` tag would equal `respond_to?(:config_files)` but for services it may be different, so +1 for tagging.

The question is how it would look like for features with more than one service such as Pulp. Do we want to create feature for each service? 

I agree it may make sense to include some module FeatureWithServices making sure the feature comply with some service related interface that we can relay on.

Reply:
Having a feature per service would be probably an overkill: the pulp feature could just return something like this:

```
def services
  { 'pulp_workers'             => 20,
    'pulp_celerybeat'          => 20,
    'pulp_resource_manager'    => 20,
    'pulp_streamer'            => 20
  }
end
```

Reply:
Thanks for the suggestions, I'll add services per feature in a module with methods to return them

Is there a way to say "what features does this machine have?" I imagine that would be helpful not only here but in the other katello scripts as well

Reply:
`ForemanMaintain.available_features` should return that. You can also pass filtering params and filter based on `class`, `label` or `tag`. So `ForemanMaintain.available_features(:tag => 'HasServices')` should do the trick.

Reply:
I tried with the config files and this worked:
```ruby
   features = ForemanMaintain.available_features(:tags => :has_config_files)
```
when the features are tagged with 
```ruby
metadata do
  tags :has_config_files
end
```

Reply:
@mbacovsky why a tag? Why not just have a method with a blank hash in `ForemanMaintain::Feature` and override it in the features with services? 

Then I could just do `ForemanMaintain.available_features.map(&:services).reduce(&:merge)`
A simliar strategy could be done for config files too

Reply:
I updated this part in my latest commit